O/openal/build_tarballs.jl

```diff
@@ -0,0 +1,56 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "openal"
+version = v"1.19.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/kcat/openal-soft.git", "96aacac10ca852fc30fd7f72f3e3c6ddbe02858c")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd openal-soft/
+cd build
+CMAKE_FLAGS="-DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DALSOFT_UTILS=false"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DALSOFT_NO_CONFIG_UTIL=true"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DALSOFT_EXAMPLES=false"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DALSOFT_TESTS=false"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DALSOFT_CONFIG=false"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DALSOFT_HRTF_DEFS=false"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DALSOFT_AMBDEC_PRESETS=false"
+cmake ${CMAKE_FLAGS} ..
+make -j${nproc}
+make install
+install_license ../COPYING 
+
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("i686", "linux"; libc=:glibc),
+    Platform("x86_64", "linux"; libc=:glibc),
+    Platform("i686", "linux"; libc=:musl),
+    Platform("x86_64", "linux"; libc=:musl),
+]
+platforms = expand_cxxstring_abis(platforms)
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libopenal", :libopenal)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="alsa_jll", uuid="45378030-f8ea-5b20-a7c7-1a9d95efb90e"))
+    Dependency(PackageSpec(name="Ogg_jll", uuid="e7412a2a-1a6e-54c0-be00-318e2571c051"))
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version=v"9")
```

Comment:
Why GCC v9?  This is not going to work with any stable Julia version.

Reply:
Just random guesses. Are those `CompilerSupportLibraries_jll` stuff documented somewhere? If not, I'd like to add docs to BB2 repo.

Reply:
What do you want to document exactly?

Reply:
A tip in the [troubleshooting section](https://juliapackaging.github.io/BinaryBuilder.jl/stable/troubleshooting/).

Reply:
A tip about _what_ exactly?

Reply:
Also, I reiterate that using GCC v9 here is a _bad_ idea, unless there is a _strong_ reason to use that, but again, this will make the package unusable in any stable Julia version

Reply:
> In the auditing phase, you may get an error like
```
libatomic.so.1: cannot open shared object file: No such file or directory
```
> a proper fix would be to add `CompilerSupportLibraries_jll` as a dependency:
```
dependencies = [
    .....
    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"))
]
```

Reply:
That's rather something to add to the auditor, that'd be something useful

Reply:
I was very confused about what you're talking about since you're commenting on the GCC line, rather than the [thread about `CompilerSupportLibraries_jll`](https://github.com/JuliaPackaging/Yggdrasil/pull/1729#discussion_r499160166)

Reply:
>Just random guesses. Are those CompilerSupportLibraries_jll stuff documented somewhere? If not, I'd like to add docs to BB2 repo.
Sorry for the confusion. The second sentence of the above reply was supposed to be put in the thread below.  

The reason is probably cultural distinctions. Chinese language is vague without contexts, when talking with each other, Chinese usually aggressively collect small pieces of related talks into a big context.  As a native speaker, I can't help assuming that other people can guess which part of the reply corresponds to which part of the context of a big topic. I didn't even notice this until you remind me in the commend above. :)

Reply:
Ok, good we've cleared the confusion!

Reply:
With https://github.com/JuliaPackaging/BinaryBuilder.jl/pull/932 the audit will also look for `libatomic` and suggest to add `CompilerSupportLibraries_jll` as dependency